### ISSUE TITLE
Discard initial frames from trajectories in pyemma.coordinates.source?

### ISSUE THREAD

**Issue by jchodera** [markovmodel/PyEMMA#897]
The idiom

``` python
import pyemma.coordinates
from glob import glob
trajectory_filenames = glob(os.path.join(source_directory, 'run*-clone*.h5'))
coordinates_source = pyemma.coordinates.source(trajectory_filenames, features=featurizer)
```

is incredibly convenient for loading datasets, but is there a simple way we can discard the first `ndiscard` frames from each trajectory to remove initial bias?

**Comment by marscher** [comment]
On 12.08.2016 22:00, John Chodera wrote:

> The idiom
> 
> import pyemma.coordinates
> from glob import glob
> trajectory_filenames = glob(os.path.join(source_directory, 'run_-clone_.h5'))
> coordinates_source = pyemma.coordinates.source(trajectory_filenames, features=featurizer)
> 
> is incredibly convenient for loading datasets, but is there a simple way
> we can discard the first |ndiscard| frames from each trajectory to
> remove initial bias?
> 
> Internally this is possible, by setting the skip parameter during 
> iteration, but we do not have a public setter this. TICA of course has 
> such a thing to set the lag time.

assume you've create reader with the source function:

``` python

Signature: reader.iterator(stride=1, lag=0, chunk=None, 
return_trajindex=True, cols=None)
Docstring:
creates an iterator to stream over the (transformed) data.

If your data is too large to fit into memory and you want to 
incrementally compute
some quantities on it, you can create an iterator on a reader or 
transformer (eg. TICA)
to avoid memory overflows.

Parameters
----------

stride : int, default=1
     Take only every stride'th frame.
lag: int, default=0
     how many frame to omit for each file.

```

**Comment by jchodera** [comment]
> Internally this is possible, by setting the skip parameter during iteration, but we do not have a public setter this.

Maybe this would be a useful feature to add?

> TICA of course has such a thing to set the lag time.

It's not quite the same as a lag time---we just want to discard the first `ndiscard` frames and then use all the frames after that, and potentially a lag time that has nothing to do with `ndiscard`.

We're working with very big datasets, of course.

**Comment by marscher** [comment]
I'm wrong. It is in the public interface ;)

see the docstring I've sent you. Note that you can pass the iterator to 
TICA.

**Comment by marscher** [comment]
uh, this is wrong... :/

even the private _estimate method, which should accept it directly, does 
not eat it.

**Comment by marscher** [comment]
fixed by #901
